shiprush.php

```diff
@@ -109,35 +113,127 @@ function __construct() {
+							$acf_repeater_child_name  = 'hiline_cold_brew_product';
+
+							for ( $i = 0; $i < ( int ) $cold_brew_qty; $i++ ) { 
+
+								$cold_brew_id = get_option( "{$acf_repeater_parent_name}_{$i}_{$acf_repeater_child_name}" );
+								
+								$excluded_products[] = ( int ) $cold_brew_id;
+							}
+						}
+						
+						if ( $excluded_products ) {
+
+							foreach ( $items as $order_item_id => $order_item ) {
+
+								if ( ! in_array( $order_item->get_product_id(), $excluded_products ) ) {
+
+									return true;
+								}
+							}
+						}
+					}
+				}
+				
+				return false;
+			}
+			
+			/*
+			 * Setup order with tracking info.
+			 *
+			 * @access public
+			 */
+			public static function set_tracking_info( $tracking_info ) {
+
+				if ( function_exists( 'wc_st_add_tracking_number' ) ) {
+
+					return wc_st_add_tracking_number( $tracking_info['order_id'], $tracking_info['tracking_number'], $tracking_info['provider'], $tracking_info['date_shipped'] );
+				}
+			}
 
 			/*
 			 * Show the meta box for shiprush button the view order page
 			 *
 			 * @access public
 			 */
-			 
 			function shiprush_meta_box() {
 			
 				global $woocommerce, $post;
 				
 				//set this variable to true if you want the plugin to hit sandbox
         		$ifsbox = USE_SANDBOX;
 				
-				$order_id=$post->ID;
-				$order = new WC_Order($order_id);
+				$order_id      = $post->ID;
+				$order         = new WC_Order( $order_id );
 				
-				 if (isset($_POST['tnum'])) 
-				 {
-					$tracking_number = $_POST['tnum'];
-					// query to update tracking number of shipment
-					update_post_meta( $OrderNumber, '_tracking_number', $tracking_number );
-					update_post_meta( $OrderNumber, '_date_shipped', time() );
-					$shipped_on=date("m/d/Y");
-					$ship_comments="Shipped on $shipped_on and tracking number is $tracking_number";
-					$change_order_status="completed";
-					$order->update_status($change_order_status, $ship_comments );
+				if ( isset( $_POST['tnum'] ) ) {
+
+					$tracking_number = sanitize_text_field( $_POST['tnum'] );
+					$shipped_on      = date("m/d/Y");
+					$ship_comments   = "Shipped on $shipped_on and tracking number is $tracking_number";
+					$tracking_info   = array(
+						'order_id'        => $order_id,
+						'tracking_number' => $tracking_number,
+						'provider'        => 'Ship Rush',
+						'date_shipped'    => $shipped_on,
+					);
+
+					// If not part shipped and a mixed products.
+					if ( 'part-shipped' != $order->get_status() && $this->is_mixed_products( $order ) ) {
+
+						$change_order_status = 'part-shipped';
+					}
 					
-                 }
+					// If mixed products and already partially shipped.
+					elseif ( $this->is_mixed_products( $order ) && 'part-shipped' == $order->get_status() ) {
+
+						$this->set_tracking_info( $tracking_info );
+						$change_order_status = 'completed';
+					}
+
+					// Not a mixed products.
+					elseif ( ! $this->is_mixed_products( $order ) ) {
```

Comment:
Not necessary to validate this since if the other conditions didn't match, you will access to this always. So it needs to be an ELSE only